plugins/module_utils/ilo_redfish_utils.py

```diff
@@ -51,6 +99,423 @@ def get_ilo_sessions(self):
         result["ret"] = True
         return result
 
+    def get_systems_data(self):
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if not response["ret"]:
+            return response
+
+        return {
+            "ret": True,
+            "msg": response
+        }
+
+    def get_managers_data(self):
+        response = self.get_request(self.root_uri + self.manager_uri)
+        if not response["ret"]:
+            return response
+
+        return {
+            "ret": True,
+            "msg": response
+        }
+
+    def get_service_bios_attributes(self):
+        # This method returns service BIOS attributes
+        response = self.get_systems_data()
+        if not response["ret"]:
+            return response
+
+        data = response["msg"]["data"]
+
+        if "Bios" not in data:
+            return {
+                "ret": False,
+                "msg": "Getting BIOS URI Failed, Key 'Bios' "
+                       "not found in %s response: %s"
+                       % (self.root_uri + self.service_root, str(data))
+            }
+
+        bios_uri = data["Bios"]["@odata.id"]
+
+        # Get service settings
+        service_uri = bios_uri + "service/settings/"
+        response = self.get_request(self.root_uri + service_uri)
+        # Check if service API doesn't support
+        if not response["ret"]:
+            service_uri = bios_uri + "oem/hpe/service/settings/"
+            response = self.get_request(self.root_uri + service_uri)
+
+            if not response["ret"]:
+                return response
+
+        return {
+            "ret": True,
+            "service_bios_attributes": response["data"]
+        }
+
+    def get_network_boot_settings(self):
+        # This method returns network boot settings present in the OOB controller
+        result = {}
+
+        uri = self.root_uri + self.service_root + self.systems_uri + "bios/"
+        response = self.get_request(uri)
+        if not response["ret"]:
+            return response
+
+        data = response["data"]
+
+        if 'Oem' in data and 'Hpe' in data["Oem"] and 'Links' in data['Oem']['Hpe'] and 'Boot' in \
+                data['Oem']['Hpe']['Links']:
+            uri = data['Oem']['Hpe']['Links']['Boot']['@odata.id'] + "settings/"
+            response = self.get_request(self.root_uri + uri)
+            if not response["ret"]:
+                return response
+            result["ret"] = True
+            self.remove_odata(response)
+            result["msg"] = self.remove_odata(response["data"])
+        else:
+            return {
+                "ret": False,
+                "msg": "Boot settings uri not found in %s response, %s" % (uri, data)
+            }
+
+        return result
+
+    def get_physical_drives_details(self):
```

Comment:
Does iLO not support standard drive resources? This functionality is all covered by standard Redfish, but it's very odd to go into OEM space for this type of info.

Reply:
Hi, i think we discussed this in the other PR [5540](https://github.com/ansible-collections/community.general/pull/5540), please refer this.

Reply:
Could we perhaps rename this method (and others touching "SmartStorage" to specifically have "SmartStorage" in the method name (such as get_smartstorage_physical_drives_details) to avoid confusion with getting drives in a generic manner?

Reply:
Changed get_physical_drives_details() to get_smartstorage_physical_drives().

Reply:
Resolving this conversation, thanks!